naga/src/valid/function.rs

```diff
@@ -399,6 +414,106 @@ impl super::Validator {
         }
         Ok(())
     }
+    fn validate_subgroup_operation(
+        &mut self,
+        op: &crate::SubgroupOperation,
+        _collective_op: &crate::CollectiveOperation,
```

Comment:
We should validate the `SubgroupOperation-CollectiveOperation` matrix.

- `Reduce` is available on all backends for all operations.
- `ExclusiveScan` & `InclusiveScan` is available on all backends only for sum and product.

Additional functionality needs to be gated behind capabilities (can be done later by interested parties).